### DISCUSSION TITLE
How exactly to compile a new variant?

### DISCUSSION THREAD

**Discussion by AidanofVT** [stm32duino/Arduino_Core_STM32#1728] [Q&A]
My board is the STM32F722ZET6. I can see this model, minus the trailing '6', in Arduino_Core_STM32/variants/STM32F7xx. However, in the Arduino IDE, no F722 boards are offered to me at all. I'm guessing that this is because the point of those 'generic' variants is to be adapted into more specific variants, and nobody has gotten around to making any F722 boards yet.

I have looked at [the instructions for making new variants,](https://github.com/stm32duino/wiki/wiki/Add-a-new-variant-(board)) and I found them unclear. I gather that I'm supposed to take the six files in the generic variant folder, make a couple alterations, add the ldscript.ld file, and compile. But compile them into what? And put the result where? If someone has an example of the exact GCC command used, or, better yet, a video of the entire process, that would be fantastic.

On a _somewhat_ related note: how cross-compatible are boards from the same family? I've been trying to upload a blink example using random F7 boards, and they aren't working at all. If that sort of thing normally _does_ work, that might mean that my problem doesn't stem from now having the exact board variant in the IDE, and it colud save me some trouble.

Any help is appreciated.

**Comment by fpistm** [comment]
If you simply want the generic, you simply have to:
- add linker script if it is not present.
- add entry in the boards.txt, you can copy the one linked to the MCU you want from the board_entry.txt to boards.txt in the GenF7 section.
- Add a custom systemCoreClock config in the generic_clock.c if it is not already defined.
